### PR TITLE
React tests

### PR DESCRIPTION
**Pull request by vadimshut**
1. Task: https://d14aa3aqnm30ti.cloudfront.net/docs/module-5/home-task/task
2. Screenshot:  
image
image
3. Deploy:-
4. Done 26.07.2022 / deadline 25.07.2022
5. Score: 30 / 30

### CODE REVIEW

File: src/components/Courses/Courses.test.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { mockedStore } from '../../common/utils-for-tests/mockStore';
+import { renderWithRouter, Wrapper } from '../../common/utils-for-tests/wrapper';
+import { BUTTON_NAMES } from '../../constants';
+import App from '../../App';
+
+jest.mock('../../services/AuthorityTokenService', () => {
+	return { authorityTokenService: { checkIsAuthorityTokenExist: () => true } };
+});
+
+describe('<Courses />', () => {
```
Comment (ArtemOsadchuck) [inline src/components/Courses/Courses.test.jsx]: Please think about human-friendly text-view of the tests
"Courses component"

File: src/components/Courses/Courses.test.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	afterEach(cleanup);
+
+	it('Courses should display amount of CourseCard equal length of courses array', () => {
```
Comment (ArtemOsadchuck) [inline src/components/Courses/Courses.test.jsx]: Please remove redundant "Courses"

File: src/components/Courses/Courses.test.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+		const { container } = render(<Courses />, { wrapper: Wrapper });
+		const cards = container.querySelectorAll('.CourseCard');
+		const cardsFromStore = mockedStore.getState().coursesReducer.courses;
+		expect(cards.length).toBe(cardsFromStore.length);
+	});
+
+	it('CourseForm should be showed after a click on a button "Add new course"', async () => {
```
Comment (ArtemOsadchuck) [inline src/components/Courses/Courses.test.jsx]: Please move this assertion to the separate "describe" scope

File: src/components/Courses/components/CourseCard/CourseCard.test.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	authorsList: [
+		{
+			name: 'author',
+			id: '9b87e8b8-6ba5-40fc-a439-c4e30a373d36',
+		},
+	],
+	id: '66cc289e-6de9-49b2-9ca7-8b4f409d6467',
+};
+
+describe('<CourseCard />', () => {
```
Comment (ArtemOsadchuck) [inline src/components/Courses/components/CourseCard/CourseCard.test.jsx]: ditto

File: src/components/Courses/components/CourseCard/CourseCard.test.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	let courseCard;
+	beforeEach(() => {
+		const { container } = render(<CourseCard {...props} />, { wrapper: Wrapper });
+		courseCard = container;
+	});
+
+	afterEach(cleanup);
```
Comment (ArtemOsadchuck) [inline src/components/Courses/components/CourseCard/CourseCard.test.jsx:28]: Please remove this action.
It's redundant now

File: src/components/Header/header.test.jsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { render, screen, cleanup } from '@testing-library/react';
+import { Header } from './Header';
+import { Wrapper } from '../../common/utils-for-tests/wrapper';
+
+describe('<Header />', () => {
```
Comment (ArtemOsadchuck) [inline src/components/Header/header.test.jsx]: ditto

File: src/components/Header/header.test.jsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+	beforeEach(() => {
+		render(<Header />, { wrapper: Wrapper });
+	});
+	afterEach(cleanup);
```
Comment (ArtemOsadchuck) [inline src/components/Header/header.test.jsx:9]: ditto

File: src/store/courses/courses.test.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+const newCourse = {
+	title: 'title',
+	description: 'description',
+	creationDate: '9/3/2021',
+	duration: 30,
+	authors: ['9b87e8b8-6ba5-40fc-a439-c4e30a373d36'],
+	id: '66cc289e-6de9-49b2-9ca7-8b4f409d6467',
+};
+
+describe('user', () => {
```
Comment (ArtemOsadchuck) [inline src/store/courses/courses.test.js:18]: typo "user"

File: src/components/Courses/Courses.test.jsx (hunk 2 above)
Comment (vadimshut) [reply src/components/Courses/Courses.test.jsx]: delete courses

File: src/components/Courses/Courses.test.jsx (hunk 3 above)
Comment (vadimshut) [reply src/components/Courses/Courses.test.jsx]: separate tests to 2 describe

File: src/components/Courses/components/CourseCard/CourseCard.test.jsx (hunk 5 above)
Comment (vadimshut) [reply src/components/Courses/components/CourseCard/CourseCard.test.jsx:28]: delete action

File: src/components/Courses/components/CourseCard/CourseCard.test.jsx (hunk 4 above)
Comment (vadimshut) [reply src/components/Courses/components/CourseCard/CourseCard.test.jsx]: rename describe from <CourseCard /> to CourseCard component and deleted external CourseCard in each test

File: src/components/Header/header.test.jsx (hunk 7 above)
Comment (vadimshut) [reply src/components/Header/header.test.jsx:9]: renamed to Header component

File: src/store/courses/courses.test.js (hunk 8 above)
Comment (vadimshut) [reply src/store/courses/courses.test.js:18]: rename test to should contain userName and userName should be exist as this test depends on header component

File: src/components/Courses/Courses.test.jsx (hunk 1 above)
Comment (vadimshut) [reply src/components/Courses/Courses.test.jsx]: renamed to Courses component

File: src/components/Header/header.test.jsx (hunk 6 above)
Comment (vadimshut) [reply src/components/Header/header.test.jsx]: renamed to Header component

**Review by ArtemOsadchuck — APPROVED** [review]

**Pull request merged**